src/libappimage/libappimage.c

```diff
@@ -1999,35 +1653,36 @@ int appimage_register_in_system(const char *path, bool verbose)
     }
 
     int type = appimage_get_type(path, verbose);
+    bool succeed = true;
 
     if(type != -1) {
 #ifdef STANDALONE
         fprintf(stderr, "\n-> Registering type %d AppImage: %s\n", type, path);
 #endif
         appimage_create_thumbnail(path, false);
+
+        char *temp_dir = desktop_integration_create_tempdir();
```

Comment:
Why does this need a tempdir at all? Can't we extract the files directly into their target location? If not, please explain this in-source. This code is entirely undocumented intention-wise.

Often, these comments describing the intention behind actions in the code are missing. I'm also not adding enough comments. Please read https://standards.mousepawmedia.com/csi.html for how it *should* be done, and try to add more comments to your code.

Reply:
Done, the idea was to reduce traversals on the AppImage. Also, it seemed simpler to implement.
And yes this is an opportunity for improving performance.

Reply:
The question was "can we extract the files directly into their target locations". As in, extract `myapp.desktop` directly into `~/.local/share/applications` etc.

Reply:
Yes, we could do it. But they have to be modified on the fly in order to avoid Desktop Environments having a hard time dealing with broken files.

Reply:
That's a great explanation, and a very good reason for this workflow. We should document that in-source, so future readers will get the idea.